essm/variables/_core.py

```diff
@@ -202,6 +215,18 @@ def collect_factor_and_basedimension(expr):
                 factor /= ifactor**count
                 dim /= idim**count
             return factor, dim
+        elif isinstance(expr, Integral):
+            # Test that integration limits have the same units
+            Variable.collect_factor_and_basedimension(sum(expr.args[1]))
```

Comment:
does it raise an exception here?

```
try:
   Variable.collect_factor_and_basedimension(sum(expr.args[1]))
except ValueError:
   raise ValueError("Integration limits must have the same units ({expr}).".format(expr=expr))
```

Reply:
I will add a test and check.

Reply:
OK, I added the try-except block with a slightly shortened error message.